src/lib/ndpi_main.c

```diff
@@ -1029,6 +1029,7 @@ static void ndpi_init_protocol_defaults(struct ndpi_detection_module_struct *ndp
 			      NDPI_PROTOCOL_APACHE_THRIFT,
 			      NDPI_PROTOCOL_JSON_RPC,
 			      NDPI_PROTOCOL_HL7,
+			      NDPI_PROTOCOL_HLS,
```

Comment:
We shouldn't need that (because we don't have a dissector for HLS; all the logic is in the HTTP code)
That should drastically simplify the diff of this PR

Reply:
> We shouldn't need that (because we don't have a dissector for HLS; all the logic is in the HTTP code) That should drastically simplify the diff of this PR

Got it, I'll remove it.